egs/voxceleb2/v2/run.sh

```diff
@@ -0,0 +1,210 @@
+  min_num_utts=8
+  awk '{print $1, NF-1}' data/voxceleb2_train_combined_no_sil/spk2utt > data/voxceleb2_train_combined_no_sil/spk2num
+  awk -v min_num_utts=${min_num_utts} '$2 >= min_num_utts {print $1, $2}' data/voxceleb2_train_combined_no_sil/spk2num | utils/filter_scp.pl - data/voxceleb2_train_combined_no_sil/spk2utt > data/voxceleb2_train_combined_no_sil/spk2utt.new
+  mv data/voxceleb2_train_combined_no_sil/spk2utt.new data/voxceleb2_train_combined_no_sil/spk2utt
+  utils/spk2utt_to_utt2spk.pl data/voxceleb2_train_combined_no_sil/spk2utt > data/voxceleb2_train_combined_no_sil/utt2spk
+
+  utils/filter_scp.pl data/voxceleb2_train_combined_no_sil/utt2spk data/voxceleb2_train_combined_no_sil/utt2num_frames > data/voxceleb2_train_combined_no_sil/utt2num_frames.new
+  mv data/voxceleb2_train_combined_no_sil/utt2num_frames.new data/voxceleb2_train_combined_no_sil/utt2num_frames
+
+  # Now we're ready to create training examples.
+  utils/fix_data_dir.sh data/voxceleb2_train_combined_no_sil
+fi
+
+if [ $stage -le 6 ]; then
+  local/nnet3/xvector/run_xvector.sh --stage $stage --train-stage -1 \
+    --data data/voxceleb2_train_combined_no_sil --nnet-dir $nnet_dir \
+    --egs-dir $nnet_dir/egs
+fi
+
+if [ $stage -le 7 ]; then
+  sid/nnet3/xvector/extract_xvectors.sh --cmd "$train_cmd --mem 4G" --nj 80 \
+    $nnet_dir data/voxceleb2_train \
+    exp/xvectors_voxceleb2_train
+
+  sid/nnet3/xvector/extract_xvectors.sh --cmd "$train_cmd --mem 4G" --nj 80 \
+    $nnet_dir data/voxceleb2_test \
+    exp/xvectors_voxceleb2_test
+fi
+
+if [ $stage -le 8 ]; then
+  local/produce_trials.py data/voxceleb2_test/utt2spk $voxceleb2_trials
+fi
+
+if [ $stage -le 9 ]; then
+  # Compute the mean vector for centering the evaluation xvectors.
+  $train_cmd exp/xvectors_voxceleb2_train/log/compute_mean.log \
+    ivector-mean scp:exp/xvectors_voxceleb2_train/xvector.scp \
+    exp/xvectors_voxceleb2_train/mean.vec || exit 1;
+
+  # This script uses LDA to decrease the dimensionality prior to PLDA.
+  lda_dim=150
+  $train_cmd exp/xvectors_voxceleb2_train/log/lda.log \
+    ivector-compute-lda --total-covariance-factor=0.0 --dim=$lda_dim \
+    "ark:ivector-subtract-global-mean scp:exp/xvectors_voxceleb2_train/xvector.scp ark:- |" \
+    ark:data/voxceleb2_train/utt2spk exp/xvectors_voxceleb2_train/transform.mat || exit 1;
+
+  # Train an out-of-domain PLDA model.
+  $train_cmd exp/xvectors_voxceleb2_train/log/plda.log \
+    ivector-compute-plda ark:data/voxceleb2_train/spk2utt \
+    "ark:ivector-subtract-global-mean scp:exp/xvectors_voxceleb2_train/xvector.scp ark:- | transform-vec exp/xvectors_voxceleb2_train/transform.mat ark:- ark:- | ivector-normalize-length ark:-  ark:- |" \
+    exp/xvectors_voxceleb2_train/plda || exit 1;
+fi
+
+if [ $stage -le 10 ]; then
+  # Treat each test utterance independently from the other test utterances of the same speaker
+  awk '{print $1, 1;}' <data/voxceleb2_test/utt2spk >exp/xvectors_voxceleb2_test/num_utts.ark
```

Comment:
I don't think this is needed, since you can just remove the --num-utts option.

Reply:
Removed.